### PR TITLE
[LoRA, Xformers] Fix xformers lora

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes https://github.com/huggingface/diffusers/issues/5135

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -310,19 +310,16 @@ def set_attention_slice(self, slice_size):
...
         self.set_processor(processor)
 
-    def set_processor(self, processor: "AttnProcessor"):
-        if (
-            hasattr(self, "processor")
-            and not isinstance(processor, LORA_ATTENTION_PROCESSORS)
-            and self.to_q.lora_layer is not None
-        ):
+    def set_processor(self, processor: "AttnProcessor", _remove_lora=False):
+        if hasattr(self, "processor") and _remove_lora and self.to_q.lora_layer is not None:
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:314]: `self.to_q.lora_layer is not None` but is it a tight check, though? What happens if someone decides to not apply LoRA to the attention layers?

**Review by sayakpaul — COMMENTED** [review]
Just one question and I am good to go. 

Maybe a test? But definitely not a blocker.

**Comment by sayakpaul** [comment]
UNet3D one seems to be flaky because I remember it from earlier PRs.

File: src/diffusers/models/attention_processor.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/diffusers/models/attention_processor.py:314]: Lora can only be removed if `self.to_q.lora_layer` is not None, so I think this is ok

**Comment by sayakpaul** [comment]
Checks passing. Merging.

**Pull request merged**